dvc/ignore.py

```diff
@@ -115,16 +115,20 @@ def __call__(self, root, dirs, files):
 
 
 class CleanTree(BaseTree):
-    def __init__(self, tree):
+    def __init__(self, tree, tree_root=None):
         self.tree = tree
+        if tree_root:
+            self._tree_root = tree_root
+        else:
+            self._tree_root = self.tree.tree_root
```

Comment:
For subrepos, CleanTree root should be the subrepo root and not the git/scm root, otherwise we will try to filter out the actual subrepo.

This was only an issue for GitTrees from brancher (WorkingTree already uses DVC repo.root_dir as the tree root)